From 96a34377dc5a0969b7b0404fce84159b7c8f89d7 Mon Sep 17 00:00:00 2001
From: Hyunchul Lee <hyc.lee@gmail.com>
Date: Thu, 1 Apr 2021 17:23:21 +0900
Subject: [PATCH] cifsd: remove calling d_path in error paths
Git-commit: 96a34377dc5a0969b7b0404fce84159b7c8f89d7
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


calling d_path is excessive in error paths.

Signed-off-by: Hyunchul Lee <hyc.lee@gmail.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/vfs.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/fs/cifsd/vfs.c b/fs/cifsd/vfs.c
index 0ecdb5121c3a..b509c90d911f 100644
--- a/fs/cifsd/vfs.c
+++ b/fs/cifsd/vfs.c
@@ -312,9 +312,8 @@ int ksmbd_vfs_read(struct ksmbd_work *work, struct ksmbd_file *fp, size_t count,
 {
 	struct file *filp;
 	ssize_t nbytes = 0;
-	char *rbuf, *name;
+	char *rbuf;
 	struct inode *inode;
-	char namebuf[NAME_MAX];
 
 	rbuf = work->aux_payload_buf;
 	filp = fp->filp;
@@ -348,11 +347,8 @@ int ksmbd_vfs_read(struct ksmbd_work *work, struct ksmbd_file *fp, size_t count,
 
 	nbytes = kernel_read(filp, rbuf, count, pos);
 	if (nbytes < 0) {
-		name = d_path(&filp->f_path, namebuf, sizeof(namebuf));
-		if (IS_ERR(name))
-			name = "(error)";
 		ksmbd_err("smb read failed for (%s), err = %zd\n",
-				name, nbytes);
+				fp->filename, nbytes);
 		return nbytes;
 	}
 
-- 
2.33.0

